sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -565,6 +565,7 @@ class annotated_ref {
     annotated_ref(T*);
     annotated_ref(const annotated_ref&) = default;
     operator T() const;
+    operator T&();
```

Comment:
Don't we need to make this const too?

Reply:
We should not as we need this to be a modifiable LValue

Reply:
removed this overload. Discussed with Roland, this approach seems error prone